### DISCUSSION TITLE
Reverse proxy support (Nginx)

### DISCUSSION THREAD

**Discussion by dc232** [alexjustesen/speedtest-tracker#937] [Q&A]
**Is your feature request related to a problem? Please describe.**
Essentially I would like a way to integrate a reverse proxy configuration into nginx so that we don't have to remember the port numbers for the application, currently, I have the following  configuration in nginx

within /home/dc/docker_configurations/nginx/config/nginx/site-confs/default.conf

```
server {
    listen 80;

    location / {
        root /config/www;  # Serve the webpage content
        index index.html;
    }

   location /speedtest/ {
        proxy_pass http://speedtest-tracker:8080;
    }

}
```

looking at the logs reveals that we are getting a 502 when reverse proxying suggesting a problem in the config 

```
192.168.9.57 - - [19/Nov/2023:09:45:18 +0000] "GET /speedtest HTTP/1.1" 301 162 "http://192.168.13.6/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.9.57 - - [19/Nov/2023:09:45:18 +0000] "GET /speedtest/ HTTP/1.1" 502 552 "http://192.168.13.6/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
```

however, within the application, there does not appear to be an option to set the base URL from the application itself to make it understand that the path to the application has now gone from

`http:${IP}:PORT`

to 

`http:${IP}:BASE_URL`

**Describe the solution you'd like**
Some documentation on how to set up the reverse proxy with the appropriate headers that need to be be passed into the application would be ideal

**Additional context**
in terms of some headers that can be passed into the application some that i have found from other applications include
```
            proxy_pass_header X-Transmission-Session-Id;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
           proxy_set_header X-Forwarded-Proto $scheme;
           proxy_set_header   X-Forwarded-Host $http_host;
           proxy_http_version 1.1;
           proxy_no_cache $cookie_session;
           proxy_http_version 1.1;
          proxy_set_header   Upgrade $http_upgrade;
          proxy_set_header   Connection keep-alive;
          proxy_cache_bypass $http_upgrade;

```

i hope the above helps in terms of determining the correct headers to pass into the configuration without dev help I don't think I can do much more than the above

**Comment by Deses** [comment]
Related #914 , (I think).

I'm reverse-proxying just fine, the issue is what you say that the app expects `http:${IP}:PORT` at some points.

**Comment by dc232** [comment]
yeah so in #914 the CORS issue comes when the system thinks its looping back on itself via the IP and PORT that's generally when CORS comes into play and the header needs to be added

**Comment by dc232** [comment]
@Deses also are you reverse proxying via nginx?

**Comment by Deses** [comment]
I'm using Nginx Proxy Manager on a docker in Unraid.

**Comment by aedelbro** [comment]
@dc232 This would be a proxy configuration issue, not the speedtest.

I would suggest doing simple host name proxy routing, ie `speedtest.mydomain.com` instead of `mydomain.com/speedtest`.

However, according to some quick searches for nginx proxy on stackoverflow / stackexchange, you need a trailing `/` on your `proxy_pass`

ie use 
```
   location /speedtest/ {
        proxy_pass http://speedtest-tracker:8080/;
    }
```

Relevant stackexchange [link](https://serverfault.com/questions/379675/nginx-reverse-proxy-url-rewrite)

**Comment by alexjustesen** [comment]
Moving this to a discussion as it's not an issue specifically with Speedtest Tracker.
